life_hc/MySQL/Items/fn_spikeStrip.sqf

```diff
@@ -1,24 +1,47 @@
-#include "\life_hc\hc_macros.hpp"
-/*
-    File: fn_spikeStrip.sqf
-    Author: Bryan "Tonic" Boardwine
-
-    This file is for Nanou's HeadlessClient.
-
-    Description:
-    This is the server-side part of it which constantly monitors the spike strip and vehicles near it.
-    First originally tried triggers but I was never any good at those nor do I like them as they
-    have a global effect.
-*/
-private ["_nearVehicles","_spikeStrip"];
-_spikeStrip = [_this,0,objNull,[objNull]] call BIS_fnc_param;
-if (isNull _spikeStrip) exitWith {}; //Bad vehicle type passed.
-
-waitUntil {_nearVehicles = nearestObjects[getPos _spikeStrip,["Car"],5]; count _nearVehicles > 0 || isNull _spikeStrip};
-
-if (isNull _spikeStrip) exitWith {}; //It was picked up?
-_vehicle = _nearVehicles select 0;
-
-if (isNil "_vehicle") exitWith {deleteVehicle _spikeStrip;};
-[_vehicle] remoteExec ["life_fnc_spikeStripEffect",_vehicle];
-deleteVehicle _spikeStrip;
\ No newline at end of file
+#include "\life_hc\hc_macros.hpp"
+/*
+    File: fn_spikeStrip.sqf
+    Author: DomT602
+
+    This file is for Nanou's HeadlessClient.
+
+    Description:
+    Adds spikestrip to hc-side array and if required - starts the monitoring of spikestrips.
+*/
+params [
+    ["_spikeStrip", objNull, [objNull]]
+];
+if (isNull _spikeStrip) exitWith {};
+
+hc_spikes pushBack _spikeStrip;
+
+if (hc_spikes isEqualTo []) then { //start monitoring spikestrips

```

Comment:
you sure about the empty array here? think you're missing a negation

Reply:
Yes, if the array is blank (no spikes are out), the loop should be started.

Reply:
You add a spikeStrip to the array 2 lines above. It will not be empty at this point, no matter what.

`count hc_spikes isEqualTo 1`